src/Guzzle/MiddleWare/RequestResponseLogger.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+namespace Acquia\ContentHubClient\Guzzle\Middleware;
+
+use Exception;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Class RequestResponseLogger
+ *
+ * @package Acquia\ContentHubClient\Guzzle\Middleware
+ */
+class RequestResponseLogger
+{
+    /**
+     * @var \Psr\Http\Message\RequestInterface
+     */
+    protected $request;
+
+    /**
+     * @var \Psr\Http\Message\ResponseInterface
+     */
+    protected $response;
+
+    /**
+     * @var \Psr\Log\LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var mixed
+     */
+    private $decodedResponseBody;
+
+    /**
+     * RequestResponseLogger constructor.
+     *
+     * @param \Psr\Http\Message\RequestInterface $request
+     * @param \Psr\Http\Message\ResponseInterface $response
+     * @param \Psr\Log\LoggerInterface $logger
+     */
+    public function __construct(RequestInterface $request, ResponseInterface $response, LoggerInterface $logger)
+    {
+        $this->request = $request;
+        $this->response = $response;
+        $this->logger = $logger;
+        $this->decodedResponseBody = json_decode($response->getBody(), true);
+    }
+
+    /**
+     * Logs response/request data.
+     *
+     * @throws \Exception
+     */
+    public function log(): void
+    {
+        if (!$this->isTrackable()) {
+            return;
+        }
+
+        $message = $this->buildLogMessage();
+
+        $this->logMessage($message, $this->response->getStatusCode());
+    }
+
+    /**
+     * Checks if a response can be tracked.
+     *
+     * @return bool
+     * @throws \Exception
+     */
+    protected function isTrackable(): bool
+    {
+        // Dont't track requests without ID.
+        if (empty($this->decodedResponseBody['request_id'])) {
+            $message = sprintf(
+                'Could not log response without ID. Response body: %s',
+                $this->response->getBody()
+            );
+            throw new Exception($message);
+        }
+
```

Comment:
It's not isTrackable's responsibility to throw an exception if request_id is empty. If a response with no request_id is not trackable, simply return false. If you need to throw an exception as well, move it to where it belongs!
Everything else looks good to me.

Reply:
Done. Thanks for review.